liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/ServerXmlDocument.java

```diff
@@ -213,18 +279,30 @@ private static boolean isValidURL(String url) {
         }
     }
    
-    public static boolean isFoundWebAppElement(NodeList nodeList) {
-         
+    private static boolean isFoundAppConfigFromAppList(NodeList nodeList) {
+        return isFoundAppConfigFromAppListByFileName(nodeList, null);
+    }
+    
+    /**
+     * @param nodeList application node list to parse.
+     * @param fileName  application file name to search, 'null' otherwise. 
+     * @return true if the application with the file name is configured.
+     */
+    private static boolean isFoundAppConfigFromAppListByFileName(NodeList nodeList, String fileName) {
+        
         boolean bFound = false; 
          
         for(int i = 0; i < nodeList.getLength(); i++) {
-            Node typeNode = nodeList.item(i).getAttributes().getNamedItem("type");
-            if (typeNode != null) {
-                String typeValue = typeNode.getNodeValue();
+            if (fileName ==  null) {
+                Node typeNode = nodeList.item(i).getAttributes().getNamedItem("type");
```

Comment:
It doesn't make sense if you will return true for a null fileName.  fileName == null is an invalid argument and you should not return true.  The logic doesn't sound right.